src/cmd/board.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2023 The ChromiumOS Authors
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+use anyhow::bail;
+use anyhow::Result;
+use argh::FromArgs;
+use glob::Pattern;
+use lium::cache::KvCache;
+use lium::repo::get_repo_dir;
+use lium::util::shell_helpers::run_bash_command;
+
+#[derive(FromArgs, PartialEq, Debug)]
+/// list supported boards
+#[argh(subcommand, name = "board")]
+pub struct Args {
+    #[argh(subcommand)]
+    nested: SubCommand,
+}
+
+pub static BOARD_CACHE: KvCache<Vec<String>> = KvCache::new("board_cache");
+static DEFAULT_BOARD_SET: &str = "board";
+
+#[derive(FromArgs, PartialEq, Debug)]
+#[argh(subcommand)]
+enum SubCommand {
+    List(ArgsList),
+}
+pub fn run(args: &Args) -> Result<()> {
+    match &args.nested {
+        SubCommand::List(args) => run_board_list(args),
+    }
+}
+
+#[derive(FromArgs, PartialEq, Debug)]
+/// List up the supporte boards
+#[argh(subcommand, name = "list")]
+pub struct ArgsList {
+    /// target cros repo directory
+    #[argh(option)]
+    repo: Option<String>,
+
+    /// glob pattern of the boards
+    #[argh(positional)]
+    boards: Option<String>,
+
+    /// only show cached list
+    #[argh(switch)]
+    cached: bool,
+}
+
+fn print_cached_boards(filter: &Pattern) -> Result<()> {
+    if let Ok(Some(boards)) = BOARD_CACHE.get(DEFAULT_BOARD_SET) {
+        for board in &boards {
+            if filter.matches(board) {
+                println!("{board}");
+            }
+        }
+        return Ok(());
+    }
+    bail!("No cache found")
+}
+
+fn update_cached_boards(repodir: &str) -> Result<()> {
+    let output = run_bash_command("cros query boards", Some(repodir))?;
+    output.status.exit_ok()?;
+    let list = String::from_utf8(output.stdout)?;
+    let boards: Vec<String> = list.lines().map(|s| s.to_string()).collect::<Vec<_>>();
+    BOARD_CACHE.set(DEFAULT_BOARD_SET, boards)?;
```

Comment:
I think the board name should be the keys of KVCache. How about creating an empty structure `struct BoardInfo` and use it as a value, instead of storing board names as a value under the single key DEFAULT_BOARD_SET?

Reply:
Ah, let me explain why, I would like to expand this command to support `--model` and `--betty` to list supported models and betty-vm boards. Thus it is using 'board' by default.

Reply:
In that case, could you add a description of a command structure of `lium board` command? It's a good idea to make sure the design sounds good before going forward.
You can edit the description of this pull-request (or file an issue) to share the design.

Maybe just using serde-json de/serializer to cache such information, instead of using KV Cache.

Reply:
OK, I decided to make this command just list up the board, and use the board name as a cahce's key. Maybe we can add some information later.